changelog/@unreleased/pr-5908.v2.yml

```diff
@@ -0,0 +1,7 @@
+type: fix
+fix:
+  description: 'Users of `PostgresConnectionConfig` will have their `ssl: true` flag
+    replaced with `sslmode: require`; this is due to a breaking change in Postgres
+    42.2.5 (https://github.com/pgjdbc/pgjdbc/issues/1307).'
```

Comment:
can we add a bit here on why this scary looking thing is actually safe? (i.e. although now true maps to verify-full, in the previous version of the driver it actually mapped to require).